ui/src/app/components/routing.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+
+import {withRoute} from 'app/routes';
+import {history} from 'app/services/history';
+
+export const Router = withRoute()(({route}) => {
+  if (!route) {
+    return <div>Not found</div>;
+  }
+  return route.render();
+});
+
+export class Redirect extends React.Component<{to: any}> {
+  componentDidMount() {
+    const {to} = this.props;
+    history.replace(to);
+  }
+
+  render() {
+    return null;
```

Comment:
Renders nothing, but navigates as a side effect. This allows us to perform a redirect by rendering this component, which is convenient (see below).